### PR TITLE
ZCS-14766: [Usage Reporting] consider the counter for COS Level

### PR DESCRIPTION
**Pull request by swathy-shaji**
Ticket - [ZCS-14766](https://synacor.atlassian.net/browse/ZCS-14766)

[ZCS-14766]: https://synacor.atlassian.net/browse/ZCS-14766?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: store/src/java/com/zimbra/cs/ldap/unboundid/UBIDLdapFilterFactory.java [hunk 1]
```diff
@@ -800,6 +814,45 @@ public ZLdapFilter accountsWithViewInHtml() {
...
+     * Accounts with Chat
+     */
+    @Override
+    public ZLdapFilter accountsWithChatAll() {
+        return new UBIDLdapFilter(
+                FilterId.ACCOUNTS_WITH_CHAT_ALL,
+                Filter.createANDFilter(
+                        FILTER_ALL_ACCOUNTS_ONLY,
+                        FILTER_NOT_SYSTEM_RESOURCE,
+                        FILTER_ACCOUNTS_WITH_SHARING));
```
Comment (RakeshAMore) [inline store/src/java/com/zimbra/cs/ldap/unboundid/UBIDLdapFilterFactory.java]: Filter with Sharing?

File: store/src/java/com/zimbra/cs/ldap/unboundid/UBIDLdapFilterFactory.java [hunk 2]
```diff
@@ -800,6 +814,45 @@ public ZLdapFilter accountsWithViewInHtml() {
...
+     * Accounts with Video
+     */
+    @Override
+    public ZLdapFilter accountsWithVideoAll() {
+        return new UBIDLdapFilter(
+                FilterId.ACCOUNTS_WITH_VIDEO_ALL,
+                Filter.createANDFilter(
+                        FILTER_ALL_ACCOUNTS_ONLY,
+                        FILTER_NOT_SYSTEM_RESOURCE,
+                        FILTER_ACCOUNTS_WITH_BRIEFCASES));
```
Comment (RakeshAMore) [inline store/src/java/com/zimbra/cs/ldap/unboundid/UBIDLdapFilterFactory.java]: Filter with Briefcase for Video?

File: soap/src/java/com/zimbra/soap/admin/type/CountObjectsType.java [hunk 3]
```diff
@@ -49,7 +49,11 @@ public enum CountObjectsType {
     internalUserAccountWithTasks(false, false),
     internalUserAccountWithSharing(false, false),
     internalUserAccountWithBriefcases(false, false),
-    internalUserAccountWithViewInHtml(false, false);
+    internalUserAccountWithViewInHtml(false, false),
+    internalUserAccountWithChatAll(false, false),
+    internalUserAccountWithVideoAll(false, false),
+    internalUserAccountWithDocumentEditing(false, false),
+    ;
```
Comment (RakeshAMore) [inline soap/src/java/com/zimbra/soap/admin/type/CountObjectsType.java]: Extra comma ?

Comment (swathy-shaji) [reply soap/src/java/com/zimbra/soap/admin/type/CountObjectsType.java]: removed

File: store/src/java/com/zimbra/cs/ldap/unboundid/UBIDLdapFilterFactory.java (hunk 1 above)
Comment (swathy-shaji) [reply store/src/java/com/zimbra/cs/ldap/unboundid/UBIDLdapFilterFactory.java]: Changed

File: store/src/java/com/zimbra/cs/ldap/unboundid/UBIDLdapFilterFactory.java (hunk 2 above)
Comment (swathy-shaji) [reply store/src/java/com/zimbra/cs/ldap/unboundid/UBIDLdapFilterFactory.java]: Changed

**Review by RakeshAMore — APPROVED** [review]

**Review by rcyarrapothu — APPROVED** [review]

**Pull request merged**